lib/Doctrine/DBAL/Connection.php

```diff
@@ -1132,7 +1133,7 @@ public function getTransactionNestingLevel()
     /**
      * Fetches the SQLSTATE associated with the last database operation.
      *
-     * @return int The last error code.
+     * @return string|bool|null The last error code.
```

Comment:
This on the other hand is not a bug, some implementors explicitly return `false` (`SQLSrvConnection::errorCode()`).

Reply:
DBAL should eliminate this discrepancy. The clients will not have to know the low level details.